pkg/controller/status/controller.go

```diff
@@ -0,0 +1,411 @@
+		timer := time.NewTicker(2 * time.Minute)
+		defer timer.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+			case <-timer.C:
+				err := limiter.Wait(ctx)
+				if err != nil {
+					klog.Errorf("Limiter error by timer: %v", err)
+				}
+			case <-c.statusCh:
+				err := limiter.Wait(ctx)
+				if err != nil {
+					klog.Errorf("Limiter error by statusMessage: %v", err)
+				}
+			}
+			if err := c.updateStatus(ctx, false); err != nil {
+				klog.Errorf("Unable to write cluster operator statusMessage: %v", err)
+			}
+		}
+	}, time.Second, ctx.Done())
+	return nil
+}
+
+func (c *Controller) updateStatus(ctx context.Context, initial bool) error {
+	existing, err := c.client.ClusterOperators().Get(ctx, c.name, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		existing = nil
+	}
+	if initial {
+		if existing != nil {
+			var reported Reported
+			if len(existing.Status.Extension.Raw) > 0 {
+				if err := json.Unmarshal(existing.Status.Extension.Raw, &reported); err != nil { //nolint: govet
+					klog.Errorf("The initial operator extension statusMessage is invalid: %v", err)
+				}
+			}
+			c.SetLastReportedTime(reported.LastReportTime.Time.UTC())
+			cs := newConditions(&existing.Status)
+			if con := cs.findCondition(configv1.OperatorDegraded); con == nil ||
+				con != nil && con.Status == configv1.ConditionFalse {
+				klog.Info("The initial operator extension statusMessage is healthy")
+			}
+		}
+	}
+
+	updated := c.merge(existing)
+	if existing == nil {
+		created, err := c.client.ClusterOperators().Create(ctx, updated, metav1.CreateOptions{}) //nolint: govet
+		if err != nil {
+			return err
+		}
+		updated.ObjectMeta = created.ObjectMeta
+		updated.Spec = created.Spec
+	} else if reflect.DeepEqual(updated.Status, existing.Status) {
+		klog.V(4).Infof("No statusMessage update necessary, objects are identical")
+		return nil
+	}
+
+	_, err = c.client.ClusterOperators().UpdateStatus(ctx, updated, metav1.UpdateOptions{})
+	return err
+}
+
+func updateDisabledAndFailingConditions(cs *conditions, ctrlStatus *controllerStatus,
+	isInitializing bool, lastTransition time.Time) {
+	switch {
+	case isInitializing:
+		// the disabled condition is optional, but set it now if we already know we're disabled
+		if ds := ctrlStatus.getStatus(DisabledStatus); ds != nil {
+			cs.setCondition(OperatorDisabled, configv1.ConditionTrue, ds.reason, ds.message, metav1.Now())
+		}
+		if !cs.hasCondition(configv1.OperatorDegraded) {
+			cs.setCondition(configv1.OperatorDegraded, configv1.ConditionFalse, "AsExpected", "", metav1.Now())
+		}
+
+	default: // once we've initialized set Failing and Disabled as best we know
+		// handle when disabled
+		if ds := ctrlStatus.getStatus(DisabledStatus); ds != nil {
+			cs.setCondition(OperatorDisabled, configv1.ConditionTrue, ds.reason, ds.message, metav1.Now())
+		} else {
+			cs.setCondition(OperatorDisabled, configv1.ConditionFalse, "AsExpected", "", metav1.Now())
+		}
```

Comment:
This if-else statement will be executed every time regardless `isInitializing` or not right? If so then we could move it out of the `switch` statement.